lib/internal/set-search-paths

```diff
@@ -26,7 +26,7 @@ _@go.set_search_paths() {
   # A plugin's own local plugin paths will appear before inherited ones. If
   # there is a version incompatibility issue with other installed plugins, this
   # allows a plugin's preferred version to take precedence.
-  @go.search_plugins '_@go.set_search_paths_add_plugin_paths'
+  @go.search_plugins '_@go.set_search_paths_add_plugin_paths' || : 
```

Comment:
`@go.search_plugins` returns 1 some times. Based on the fact that without strict mode, the script executes without errors, I guess that when it returns 1, it is not an actual error. It is probably meant to help when `@go.search_plugin` is used in an if statement. 

If the above is correct, `@go.seach_plugin` is used in an if statement only once in the whole project, while it is used 8 times as here. Should the function be redesigned or, if the above 'fix' is sufficient, should the same 'fix' be applied in all other 7 case?